include/nuttx/mm/map.h

```diff
@@ -80,4 +80,137 @@ struct mm_map_s
  * Public Function Prototypes
  ****************************************************************************/
 
-#endif /* __INCLUDE_NUTTX_MM_MM_MAP_H */
+#ifdef CONFIG_MM_MAP
+
+/****************************************************************************
+ * Name: mm_map_initialize
+ *
+ * Description:
+ *   Initialization function, called only by group_initialize
+ *
+ * Input Parameters:
+ *   mm - Pointer to the mm_map structure to be initialized
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void mm_map_initialize(FAR struct mm_map_s *mm);
+
+/****************************************************************************
+ * Name: mm_map_destroy
+ *
+ * Description:
+ *   Uninitialization function, called only by group_release
+ *
+ * Input Parameters:
+ *   mm - Pointer to the mm_map structure to be initialized
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void mm_map_destroy(FAR struct mm_map_s *mm);
+
+/****************************************************************************
+ * Name: mm_map_add
+ *
+ * Description:
+ *   Adds a virtual memory area into the list of mappings
+ *
+ * Input Parameters:
+ *   entry - A pointer to mm_map_entry_s, mapping info to be added
+ *
+ * Returned Value:
+ *   OK        Added succesfully
+ *   -EINVAL:  Invalid attempt to get the semaphore
+ *   -EINTR:   The wait was interrupted by the receipt of a signal.
+ *   -ENOMEM:  Out of memory
+ *
+ ****************************************************************************/
+
+int mm_map_add(FAR struct mm_map_entry_s *entry);
```

Comment:
it's better to add task_group_s as an explicit argument since it's very use to create a mapping for other (not current) process sometime.

Reply:
I think that it would be very strange use case. With real MMU you'd have to switch address environments to do that mapping. In which case you could map memory for some other process?

Reply:
If you want to do some advance IPC, it's a performance boost if we can do map from source process to the target process, but since some function require task_group_s as the first argument, it's good to keep consistence that all function accept task_group_s  as the first argument.

Reply:
A process can never inject new mappings to another process' address space. That wouldn't make sense. The only reason why kernel internal munmap has it here is because group can't be used during process exit. So we are passing NULL during process exit

can you give an example of a (posix) interface which would require mapping to another process' address space? I am not aware of any OS that would do that, even inside kernel.

Reply:
If you insist that we add the group as a parameter here, I can do. It just can't be anything else than currently running one IMHO.

Reply:
passing mm_map_s and mm_map_entry_s in places where needed.